src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultEnumConverter.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.Json.Serialization.Policies;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal sealed class DefaultEnumConverter<TValue> : JsonValueConverter<TValue>
+    {
+        public bool TreatAsString { get; private set; }
+
+        internal DefaultEnumConverter(bool treatAsString)
+        {
+            TreatAsString = treatAsString;
+        }
+
+        public override bool TryRead(Type valueType, ref Utf8JsonReader reader, out TValue value)
+        {
+            if (TreatAsString)
+            {
+                // Assume the token is a string
+                if (reader.TokenType != JsonTokenType.String)
+                {
+                    value = default;
+                    return false;
+                }
+
+#if !BUILDING_INBOX_LIBRARY 
+                // todo: add code here to handle NS2.0
+                throw new NotImplementedException(SR.EnumConverterNotImplemented);
+#else
+                string enumString = reader.GetString();
+                if (!Enum.TryParse(valueType, enumString, out object objValue))
+                {
+                    value = default;
+                    return false;
+                }
+
+                value= (TValue)objValue;
+                return true;
+#endif
+            }
+
+            if (reader.TokenType != JsonTokenType.Number)
+            {
+                value = default;
+                return false;
+            }
+
+            if (!reader.TryGetUInt64(out ulong ulongValue))
+            {
+                value = default;
+                return false;
+            }
+
+            value = (TValue)Enum.ToObject(valueType, ulongValue);
+            return true;
+        }
+
+        public override void Write(TValue value, ref Utf8JsonWriter writer)
+        {
+            if (TreatAsString)
+            {
+                writer.WriteStringValue(value.ToString());
+            }
+            else
+            {
+                Type underlyingType = Enum.GetUnderlyingType(value.GetType());
+
+                if (underlyingType == typeof(ulong))
+                {
+                    // Keep +sign
```

Comment:
Can you explain what you mean by this? What `+` sign?

Reply:
I changed the comment to "Use the UInt64 converter to prevent conversion into a signed value."